packages/core/src/utils/security/handlers/apiKey.ts

```diff
@@ -0,0 +1,38 @@
+import { fromNullable, getOrElse, mapNullable } from 'fp-ts/lib/Option';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { get } from 'lodash';
+import { SecurityScheme } from './types';
+import { when } from './utils';
+
+export const apiKeyInCookie = {
+  test: ({ type, in: where }: SecurityScheme) => where === 'cookie' && type === 'apiKey',
+  handle: <R, I>(someInput: I, name: string, resource?: R) => {
+    const probablyCookie = get(someInput, ['headers', 'cookie']);
+
+    const isApiKeyInCookie = pipe(
+      fromNullable(probablyCookie),
+      mapNullable(cookie => new RegExp(`${name}=.+`).test(cookie)),
```

Comment:
Not sure about this regexp. Have you tried some weird edge cases? 
`Set-Cookie: id=a3fWa; Expires=Wed, 21 Oct 2015 07:28:00 GMT;`
Or
`Set-Cookie: id=bla=bla;`

Reply:
ha! I haven't, thanks, will look into that

Reply:
@chris-miaskowski, after giving it a bit of thought, I am inclined to say that this regex is OK.
Also, I assumed that `Set-Cookie` was mistaken for `Cookie` (?). If not, please LMK :)

In the end what's happening is this:
```
new RegExp('SOME_COOKIE=.+').test('SOME_COOKIE=<<some_value>>)
```
So, I'm just checking if there is this SOME_COOKIE with some (any) value. The value itself is not important, it's only about the presence of it (where `it` is a cookie named `name`) . So, the examples you provided, would pass the test and so would any other value for `id` cookie. 

Also, I think that `Expires` is kinda irrelevant, since it wouldn't be present in `Cookie` header (?). And even it it was present in `Cookie` header it would be (in the context of the example) considered another cookie named `Expires` I think.

I don't know if my answer addresses your concerns, LMK :)